src/pages/RoomMembersPage.js

```diff
@@ -0,0 +1,335 @@
+            Navigation.goBack(ROUTES.REPORT_WITH_ID_DETAILS.getRoute(props.report.reportID));
+            return;
+        }
+        getRoomMembers();
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, []);
+
+    /**
+     * Open the modal to invite a user
+     */
+    const inviteUser = () => {
+        setSearchValue('');
+        Navigation.navigate(ROUTES.ROOM_INVITE.getRoute(props.report.reportID));
+    };
+
+    /**
+     * Remove selected users from the room
+     */
+    const removeUsers = () => {
+        Report.removeFromRoom(props.report.reportID, selectedMembers);
+        setSelectedMembers([]);
+        setRemoveMembersConfirmModalVisible(false);
+    };
+
+    /**
+     * Add user from the selectedMembers list
+     *
+     * @param {String} login
+     */
+    const addUser = useCallback((accountID) => {
+        setSelectedMembers((prevSelected) => [...prevSelected, accountID]);
+    }, []);
+
+    /**
+     * Remove user from the selectedEmployees list
+     *
+     * @param {String} login
+     */
+    const removeUser = useCallback((accountID) => {
+        setSelectedMembers((prevSelected) => _.without(prevSelected, accountID));
+    }, []);
+
+    /**
+     * Toggle user from the selectedMembers list
+     *
+     * @param {String} accountID
+     * @param {String} pendingAction
+     *
+     */
+    const toggleUser = useCallback(
+        (accountID, pendingAction) => {
+            if (pendingAction === CONST.RED_BRICK_ROAD_PENDING_ACTION.DELETE) {
+                return;
+            }
+
+            // Add or remove the user if the checkbox is enabled
+            if (_.contains(selectedMembers, Number(accountID))) {
+                removeUser(Number(accountID));
+            } else {
+                addUser(Number(accountID));
+            }
+        },
+        [selectedMembers, addUser, removeUser],
+    );
+
+    /**
+     * Add or remove all users passed from the selectedMembers list
+     * @param {Object} memberList
+     */
+    const toggleAllUsers = (memberList) => {
+        const enabledAccounts = _.filter(memberList, (member) => !member.isDisabled);
+        const everyoneSelected = _.every(enabledAccounts, (member) => _.contains(selectedMembers, Number(member.keyForList)));
+
+        if (everyoneSelected) {
+            setSelectedMembers([]);
+        } else {
+            const everyAccountId = _.map(enabledAccounts, (member) => Number(member.keyForList));
+            setSelectedMembers(everyAccountId);
+        }
+    };
+
+    /**
+     * Show the modal to confirm removal of the selected members
+     */
+    const askForConfirmationToRemove = () => {
+        setRemoveMembersConfirmModalVisible(true);
+    };
+
+    const getMemberOptions = () => {
+        let result = [];
+
+        _.each(props.report.participantAccountIDs, (accountID) => {
+            const details = props.personalDetails[accountID];
+
+            if (!details) {
```

Comment:
We also need to filter out the optimistic personal detail here to prevent [this bug](https://github.com/Expensify/App/issues/45898).
More detail in [this comment](https://github.com/Expensify/App/issues/45898#issuecomment-2242928643)